### PR TITLE
IS-523: Support fee2.0 APIs

### PR DESCRIPTION
**Pull request by boyeon555**
- Add DepositTransaction and DepositTransactionBuilder
  - Applied to sign
  - Adds validation of deposit transaction
- Add test cases to validate deploy transaction
 
  - test cases for deposit in detail
    -  getting transaction result correctly
    - unit and integration tests for adding and withdrawing a deposit
    - revising setting data by adding id value for withdrawing
    - signing with a deposit transaction
    - generating deposit transaction by its builder

### CODE REVIEW

File: iconsdk/builder/transaction_builder.py [hunk 1]
```diff
@@ -348,3 +375,45 @@ def from_dict(cls, transaction_as_dict: dict) -> 'MessageTransactionBuilder':
...
+        self._action = action
+        return self
+
+    def id(self, id: str):
+        self._id = id
+        return self
+
+    def build(self) -> DepositTransaction:
+        if not self._action:
+            raise DataTypeException("'action' should be required. Available values for it are 'add' or 'withdraw'.")
```
Comment (sink772) [inline iconsdk/builder/transaction_builder.py]:
```suggestion
            raise DataTypeException("'action' should be provided. Available values are 'add' or 'withdraw'.")
```

File: iconsdk/builder/transaction_builder.py [hunk 2]
```diff
@@ -348,3 +375,45 @@ def from_dict(cls, transaction_as_dict: dict) -> 'MessageTransactionBuilder':
...
+        if self._action == "withdraw":
+            if not self._id:
+                raise DataTypeException("When action is 'withdraw', 'id' should be required.")
```
Comment (sink772) [inline iconsdk/builder/transaction_builder.py]:
```suggestion
                raise DataTypeException("When action is 'withdraw', 'id' should be provided.")
```

File: iconsdk/builder/transaction_builder.py [hunk 3]
```diff
@@ -348,3 +375,45 @@ def from_dict(cls, transaction_as_dict: dict) -> 'MessageTransactionBuilder':
...
+            if self._value:
+                raise DataTypeException("When action is 'withdraw'. 'value' should not be required.")
```
Comment (sink772) [inline iconsdk/builder/transaction_builder.py]:
```suggestion
                raise DataTypeException("When action is 'withdraw'. 'value' should not be provided.")
```

File: iconsdk/builder/transaction_builder.py [hunk 4]
```diff
@@ -348,3 +375,45 @@ def from_dict(cls, transaction_as_dict: dict) -> 'MessageTransactionBuilder':
...
+        if self._action == "add" and not self._value:
+            raise DataTypeException("When action is 'add', 'value' should be required.")
```
Comment (sink772) [inline iconsdk/builder/transaction_builder.py]:
```suggestion
            raise DataTypeException("When action is 'add', 'value' should be provided.")
```

File: README.md [hunk 5]
```diff
@@ -1016,6 +1040,57 @@ transaction = MessageTransactionBuilder()\
...
+
+Builder for **DepositTransaction** object
+
+#### Methods
+
+- from_ : The wallet address making a transaction. The default address is your account address.
+- to : SCORE address to receive a transaction
+- value : The amount of ICX to be deposited. It is used only for 'add' action. (Optional)
+- action : "add" or "withdraw".
+- Id : Transaction hash prefixed with '0x'. It is used only for 'withdraw' action. (Optional)
```
[the remark this answers is not in the document]
Comment (boyeon555) [reply README.md]: It applies to the code

File: .travis.yml [hunk 6]
```diff
@@ -6,7 +6,7 @@ services:
   - docker
 
 install:
-- docker run -d -it --name tbears-container -p 9000:9000 iconloop/tbears:1.2.0-rc2
+- docker run -d -it --name tbears-container -p 9000:9000 iconloop/tbears:1.3.0-rc1
```
Comment (boyeon555) [inline .travis.yml]:
```suggestion
- docker run -d -it --name tbears-container -p 9000:9000 iconloop/tbears:1.3.0-rc
```

File: .travis.yml [hunk 7]
```diff
@@ -6,7 +6,7 @@ services:
   - docker
```
Comment (boyeon555) [inline .travis.yml]:
```suggestion
  - docker -h
```

**Pull request merged**